### PR TITLE
Changes for GRR packages

### PR DESCRIPTION
**Pull request by Onager**
Changes for GRR packages

### CODE REVIEW

File: l2tdevtools/build_helpers/dpkg.py [hunk 1]
```diff
@@ -186,11 +190,17 @@ def _CreateOriginalSourcePackageFromZip(
             zip_info = zip_file.getinfo(filename)
             tar_info = tarfile.TarInfo(filename)
             tar_info.size = zip_info.file_size
+            date_time = zip_info.date_time
+            mtime = datetime.datetime(
```
Comment (joachimmetz) [inline l2tdevtools/build_helpers/dpkg.py]: Maybe this short hand works for you?
```
mtime = datetime.datetime(*date_time)
```

File: l2tdevtools/build_helpers/dpkg.py [hunk 2]
```diff
@@ -186,11 +190,17 @@ def _CreateOriginalSourcePackageFromZip(
             zip_info = zip_file.getinfo(filename)
             tar_info = tarfile.TarInfo(filename)
             tar_info.size = zip_info.file_size
+            date_time = zip_info.date_time
```
Comment (joachimmetz) [inline l2tdevtools/build_helpers/dpkg.py]: IMHO it is worth a comment to what you're accomplishing here and why

File: l2tdevtools/build_helpers/dpkg.py [hunk 3]
```diff
@@ -265,7 +275,9 @@ def _RemoveOlderDPKGPackages(self, project_name, project_version):
         logging.info('Removing: {0:s}'.format(filename))
         os.remove(filename)
 
-  def _RemoveOlderOriginalSourcePackage(self, project_name, project_version):
+  def _RemoveOlderOriginalSourcePackage(
+      self, project_name, project_version, version_suffix=None,
+      distribution=None):
     """Removes previous versions of original source package.
 
     Args:
```
Comment (joachimmetz) [inline l2tdevtools/build_helpers/dpkg.py]: Update Args docstring.

File: l2tdevtools/build_helpers/dpkg.py [hunk 4]
```diff
@@ -274,11 +286,17 @@ def _RemoveOlderOriginalSourcePackage(self, project_name, project_version):
     """
     filenames_to_ignore = '^{0:s}_{1!s}.orig.tar.gz'.format(
         project_name, project_version)
+    if version_suffix and distribution:
```
Comment (joachimmetz) [inline l2tdevtools/build_helpers/dpkg.py]: I opt to make this an if-else instead of an set-if-override

File: l2tdevtools/build_helpers/dpkg.py [hunk 5]
```diff
@@ -274,11 +286,17 @@ def _RemoveOlderOriginalSourcePackage(self, project_name, project_version):
...
+      filenames_to_ignore = '{0:s}_{1:s}{2:s}~{3:s}.orig.tar.gz'.format(
+          project_name, project_version, version_suffix, distribution)
     filenames_to_ignore = re.compile(filenames_to_ignore)
 
     # Remove files of previous versions in the format:
     # project_version.orig.tar.gz
     filenames_glob = '{0:s}_*.orig.tar.gz'.format(project_name)
```
Comment (joachimmetz) [inline l2tdevtools/build_helpers/dpkg.py]: I opt to make this an if-else instead of an set-if-override

File: l2tdevtools/build_helpers/dpkg.py [hunk 6]
```diff
@@ -525,7 +541,8 @@ def Build(self, source_helper_object):
     exit_code = subprocess.call('(cd {0:s} && {1:s})'.format(
         source_directory, command), shell=True)
     if exit_code != 0:
-      logging.error('Running: "{0:s}" failed.'.format(command))
+      logging.error('Running: "(cd {0:s} && {1:s}" failed.'.format(
```
Comment (joachimmetz) [inline l2tdevtools/build_helpers/dpkg.py]: Maybe change to: `Failed to run ... with exit code: ...`

from the current wording it is not directly clear things failed.

File: l2tdevtools/build_helpers/dpkg.py [hunk 7]
```diff
@@ -810,8 +825,8 @@ def Build(self, source_helper_object):
     exit_code = subprocess.call('(cd {0:s} && {1:s})'.format(
         source_directory, command), shell=True)
     if exit_code != 0:
-      logging.error('Running: "{0:s}" failed.'.format(command))
-      return False
+      logging.error('Running: "(cd {0:s} && {1:s}" failed.'.format(
```
Comment (joachimmetz) [inline l2tdevtools/build_helpers/dpkg.py]: Maybe change to: Failed to run ... with exit code: ...

File: l2tdevtools/build_helpers/dpkg.py [hunk 8]
```diff
@@ -847,7 +862,8 @@ def Clean(self, source_helper_object):
     project_version = source_helper_object.GetProjectVersion()
 
     self._RemoveOlderOriginalSourcePackage(
-        source_helper_object.project_name, project_version)
+        source_helper_object.project_name, project_version,
+        self.version_suffix, self.distribution)
```
Comment (joachimmetz) [inline l2tdevtools/build_helpers/dpkg.py]: additional args are kwargs not positional args, please adjust call to method

File: l2tdevtools/build_helpers/dpkg.py (hunk 2 above)
Comment (Onager) [reply l2tdevtools/build_helpers/dpkg.py]: Done

File: l2tdevtools/build_helpers/dpkg.py (hunk 1 above)
Comment (Onager) [reply l2tdevtools/build_helpers/dpkg.py]: Done

File: l2tdevtools/build_helpers/dpkg.py (hunk 3 above)
Comment (Onager) [reply l2tdevtools/build_helpers/dpkg.py]: Done

File: l2tdevtools/build_helpers/dpkg.py (hunk 4 above)
Comment (Onager) [reply l2tdevtools/build_helpers/dpkg.py]: Done

File: l2tdevtools/build_helpers/dpkg.py (hunk 5 above)
Comment (Onager) [reply l2tdevtools/build_helpers/dpkg.py]: Done

File: l2tdevtools/build_helpers/dpkg.py (hunk 6 above)
Comment (Onager) [reply l2tdevtools/build_helpers/dpkg.py]: Done

File: l2tdevtools/build_helpers/dpkg.py (hunk 7 above)
Comment (Onager) [reply l2tdevtools/build_helpers/dpkg.py]: Done

**Pull request merged**